source/cgal/polygon.cc

```diff
@@ -0,0 +1,171 @@
+
+#  include <CGAL/Boolean_set_operations_2.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+#  ifndef DOXYGEN
+// Template implementations
+namespace CGALWrappers
+{
+  template <typename KernelType>
+  void
+  dealii_cell_to_cgal_polygon(
+    const typename Triangulation<2, 2>::cell_iterator &cell,
+    const Mapping<2, 2>                               &mapping,
+    CGAL::Polygon_2<KernelType>                       &polygon)
+  {
+    const auto &vertices = mapping.get_vertices(cell);
+    polygon.clear();
+    if (cell->reference_cell() == ReferenceCells::Triangle)
+      {
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[0]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[1]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[2]));
+      }
+    else if (cell->reference_cell() == ReferenceCells::Quadrilateral)
+      {
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[0]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[1]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[3]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[2]));
+      }
+    else
+      {
+        DEAL_II_ASSERT_UNREACHABLE();
+      }
+  }
+
+
+
+  template <typename KernelType>
+  void
+  dealii_tria_to_cgal_polygon(const Triangulation<2, 2>   &tria,
+                              CGAL::Polygon_2<KernelType> &polygon)
+  {
+    std::map<unsigned int, unsigned int> face_vertex_indices;
+
+    for (const auto &cell : tria.active_cell_iterators())
+      {
+        for (const unsigned int i : cell->face_indices())
+          {
+            const typename Triangulation<2, 2>::face_iterator &face =
+              cell->face(i);
+            if (face->at_boundary())
+              {
+                if (cell->reference_cell() == ReferenceCells::Quadrilateral &&
+                    (i == 0 || i == 3))
+                  {
+                    face_vertex_indices[face->vertex_index(1)] =
+                      face->vertex_index(0);
+                  }
+                else
+                  {
+                    face_vertex_indices[face->vertex_index(0)] =
+                      face->vertex_index(1);
+                  }
```

Comment:
Can you add a comment here that explains that what the `if` does is ensure that you are adding boundary lines in an order so that they go counter-clockwise? That may not be obvious to a reader.